### PR TITLE
#5139 Restart workspace fail need highlight the unstable workspace

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist

- [x] [Review Contribution Guidelines](https://oharastream.github.io/en/docs/master/contributing/)
- [x] Add at least one reviewer (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add an assignee (of course, you are the assignee by default)
- [x] Do not set milestone, projects and labels (PR labels are added automatically by Jenkins)
- [x] Link to an issue
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (If the failed tests are unrelated to this PR, add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Workspace/Settings/DangerZone/RestartWorkspace.js [hunk 1]
```diff
@@ -116,7 +126,6 @@ const RestartWorkspace = (props) => {
     const startWorker = {
       name: 'start worker',
       action: () => startWorkerAction(worker.name),
-      revertAction: () => stopWorkerAction(worker.name),
```
Comment (wu87988622) [inline ohara-manager/client/src/components/Workspace/Settings/DangerZone/RestartWorkspace.js:119]: 這個方法永遠不會走到...

File: ohara-manager/client/cypress/it/navigator/dangerZone.test.ts [hunk 2]
```diff
@@ -1227,4 +1227,38 @@ describe('Danger Zone', () => {
 
     cy.findByText('CLOSE').parent('button').should('be.enabled').click();
   });
+
+  it('restart workspace is marked as unstable when it fails', () => {
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator/dangerZone.test.ts:1231]: could rephrase this to _should mark the workspace as unstable when it fails_ :)

File: ohara-manager/client/cypress/it/navigator/dangerZone.test.ts [hunk 3]
```diff
@@ -1227,4 +1227,38 @@ describe('Danger Zone', () => {
...
+        cy.findByText('RESTART').click();
+      });
+
+    cy.findByText('ERROR', { exact: false });
+
+    cy.findByText('CLOSE').parent('button').should('be.enabled').click();
+
+    cy.findByText('ABORT').parent('button').click();
+
+    cy.visit('/');
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator/dangerZone.test.ts:1257]: Could you add some comment to better convey our intent? It's not very clear to me why this visit is necessary

File: ohara-manager/client/cypress/it/navigator/dangerZone.test.ts [hunk 4]
```diff
@@ -1227,4 +1227,38 @@ describe('Danger Zone', () => {
...
+    });
+
+    cy.findAllByRole('dialog')
+      .filter(':visible')
+      .should('have.length', 1)
+      .within(() => {
+        cy.findByText('RESTART').click();
+      });
+
+    cy.findByText('ERROR', { exact: false });
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/navigator/dangerZone.test.ts:1251]: We should probably assert here, i know if would fail before the `should` but this is how we currently writing all our tests

File: ohara-manager/client/cypress/it/navigator/dangerZone.test.ts (hunk 2 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.test.ts:1231]: thanks

File: ohara-manager/client/cypress/it/navigator/dangerZone.test.ts (hunk 3 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.test.ts:1257]: ok

File: ohara-manager/client/cypress/it/navigator/dangerZone.test.ts (hunk 4 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/it/navigator/dangerZone.test.ts:1251]: ok thanks

**Comment by wu87988622** [comment]
retry manager-it

**Comment by wu87988622** [comment]
retry manager-it

**Comment by wu87988622** [comment]
retry manager-it

**Comment by wu87988622** [comment]
retry
